### PR TITLE
 lookup the coin records in bulk

### PR DESCRIPTION
**Pull request by arvidn**
When adding coins to the wallet. This shaves off about 9% from sync-time the toilet wallet:

| operation | before | after | after / before |
| --- | --- | --- | --- |
| wallet sync | 89.77s | 81.85s | 91.2% |

### CODE REVIEW

File: chia/wallet/wallet_state_manager.py [hunk 1]
```diff
@@ -876,12 +876,16 @@ async def new_coin_state(
         used_up_to = -1
         ph_to_index_cache: LRUCache = LRUCache(100)
 
-        for coin_state in coin_states:
+        local_records: List[Optional[WalletCoinRecord]] = await self.coin_store.get_coin_records(
+            [st.coin.name() for st in coin_states]
```
Comment (arvidn) [inline chia/wallet/wallet_state_manager.py:880]: this makes us compute the coin name twice. It's still a net improvement though, and there's no simple way of avoiding it. The `WalletCoinRecord` does not contain a name field.

File: chia/wallet/wallet_state_manager.py [hunk 2]
```diff
@@ -876,12 +876,16 @@ async def new_coin_state(
...
+        )
+
+        assert len(local_records) == len(coin_states)
+        for coin_state, local_record in zip(coin_states, local_records):
             coin_name: bytes32 = coin_state.coin.name()
             wallet_info: Optional[Tuple[uint32, WalletType]] = await self.get_wallet_id_for_puzzle_hash(
                 coin_state.coin.puzzle_hash
             )
-            local_record: Optional[WalletCoinRecord] = await self.coin_store.get_coin_record(coin_name)
```
Comment (arvidn) [inline chia/wallet/wallet_state_manager.py:884]: this is the lookup we're saving

File: chia/wallet/wallet_state_manager.py [hunk 3]
```diff
@@ -967,10 +971,10 @@ async def new_coin_state(
                     )
                     await self.coin_store.add_coin_record(record)
                     # Coin first received
-                    coin_record: Optional[WalletCoinRecord] = await self.coin_store.get_coin_record(
+                    parent_coin_record: Optional[WalletCoinRecord] = await self.coin_store.get_coin_record(
                         coin_state.coin.parent_coin_info
                     )
-                    if coin_record is not None and wallet_type.value == coin_record.wallet_type:
+                    if parent_coin_record is not None and wallet_type.value == parent_coin_record.wallet_type:
```
Comment (arvidn) [inline chia/wallet/wallet_state_manager.py:977]: this is an unrelated renaming. I think this is a better na,e

**Review by trepca — APPROVED** [review]

File: chia/wallet/wallet_state_manager.py (hunk 1 above)
Comment (trepca) [reply chia/wallet/wallet_state_manager.py:880]: you could build a dict lookup for names, but probably marginal
